packages/transport/src/modules/relationships/local/AuditLog.ts

```diff
@@ -0,0 +1,50 @@
+import { serialize, type, validate } from "@js-soft/ts-serval";
+import _ from "lodash";
+import { CoreAddress, CoreDate, CoreSerializable, ICoreAddress, ICoreDate } from "../../../core";
+import { AuditLogEntryReason, AuditLog as BackboneAuditLog } from "../transmission/AuditLog";
+import { RelationshipStatus } from "../transmission/RelationshipStatus";
+
+export interface IAuditLogEntry {
+    createdAt: ICoreDate;
+    createdBy: ICoreAddress;
+    reason: AuditLogEntryReason;
+    oldStatus?: RelationshipStatus;
+    newStatus: RelationshipStatus;
+}
+
+@type("AuditLogEntry")
+export class AuditLogEntry extends CoreSerializable implements IAuditLogEntry {
+    @validate()
+    @serialize()
+    public createdAt: CoreDate;
+
+    @validate()
+    @serialize()
+    public createdBy: CoreAddress;
+
+    @validate()
+    @serialize()
+    public reason: AuditLogEntryReason;
+
+    @validate({ nullable: true })
+    @serialize()
+    public oldStatus?: RelationshipStatus;
+
+    @validate()
+    @serialize()
+    public newStatus: RelationshipStatus;
+
+    public static from(value: IAuditLogEntry): AuditLogEntry {
+        return this.fromAny({ ...value, oldStatus: value.oldStatus ?? undefined });
+    }
+}
+
+export class AuditLog {
+    public static fromBackboneAuditLog(backboneAuditLog: BackboneAuditLog): AuditLogEntry[] {
+        const auditLog: AuditLogEntry[] = [];
+        backboneAuditLog.forEach((entry) =>
+            auditLog.push(AuditLogEntry.from({ ...entry, createdAt: CoreDate.from(entry.createdAt), createdBy: CoreAddress.from(entry.createdBy) }))
+        );
```

Comment:
why not using `.map()` ? Additionally `createdAt: CoreDate.from(entry.createdAt), createdBy: CoreAddress.from(entry.createdBy)` should not be nessecary, this is ts-servals job

Reply:
I didn't know about `.map` XD - and I get an error without `createdAt, createdBy`

Reply:
In that case I would argue just mapping everything by hand

Reply:
I don't understand what you mean

Reply:
```
const auditLog = backboneAuditLog.map((entry) => {
            return AuditLogEntry.from({
                createdAt: CoreDate.from(entry.createdAt),
                createdBy: CoreAddress.from(entry.createdBy),
                createdByDevice: CoreId.from(entry.createdByDevice),
                reason: entry.reason,
                oldStatus: entry.oldStatus,
                newStatus: entry.newStatus
            });
        });

        return _.orderBy(auditLog, ["createdAt"], ["asc"]);
```

would be like this. btw - you missed `createdByDevice` in AuditLogEntry